File: Back-end/src/main/java/br/jus/stj/siscovi/model/ListaTotalMensalData.java
package br.jus.stj.siscovi.model;

import java.sql.Date;
import java.util.ArrayList;

public class ListaTotalMensalData {
    private Date dataReferencia;
    private ArrayList<TotalMensal> totais;

    public ListaTotalMensalData(Date dataReferencia, ArrayList<TotalMensal> totais) {
        this.dataReferencia = dataReferencia;
        this.totais = totais;
    }
    public int getTotaisSize() {
        return totais.size();
    }

    public Date getDataReferencia() {
        return dataReferencia;
    }

    @Override
    public String toString() {
        return "ListaTotalMensalData{" +
                "dataReferencia=" + dataReferencia +
                ", totais=" + totais +
                '}';
    }
}

File: Front-end/src/app/totalMensal/lista-total-mensal-data.ts
import {TotalMensal} from './totalMensal';

export class ListaTotalMensalData {
    private _dataReferencia: Date;
    private _totais: TotalMensal[];


    constructor(dataReferencia: Date, totais: TotalMensal[]) {
        this._dataReferencia = dataReferencia;
        this._totais = totais;
    }

    get dataReferencia(): Date {
        return this._dataReferencia;
    }

    get totais(): TotalMensal[] {
        return this._totais;
    }
}
